src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -4827,4 +4857,201 @@ static class D
+              IL_0010:  ret
+            }
+            """);
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/70841")]
+    public void InterceptorStructBaseMethod()
+    {
+        var program = ("""
+            using System;
+
+            MyStruct value = default;
+            Console.WriteLine(value.Equals((object)1));
+
+            public struct MyStruct { }
+            """, "Program.cs");
+
+        var interceptor = ("""
+            using System.Runtime.CompilerServices;
+
+            namespace MyInterceptors
+            {
+                public static class Interceptors
+                {
+                    [InterceptsLocation(@"Program.cs", 4, 25)]
+                    public static bool Equals(this System.ValueType value, object other) => true;
+                }
+            }
+            """, "Interceptor.cs");
+
+        var verifier = CompileAndVerify(new[] { program, s_attributesSource }, parseOptions: RegularWithInterceptors, expectedOutput: "False");
+        verifier.VerifyDiagnostics();
+        verifier.VerifyIL("<top-level-statements-entry-point>", """
+            {
+              // Code size       33 (0x21)
+              .maxstack  2
+              .locals init (MyStruct V_0) //value
+              IL_0000:  ldloca.s   V_0
+              IL_0002:  initobj    "MyStruct"
+              IL_0008:  ldloca.s   V_0
+              IL_000a:  ldc.i4.1
+              IL_000b:  box        "int"
+              IL_0010:  constrained. "MyStruct"
+              IL_0016:  callvirt   "bool object.Equals(object)"
+              IL_001b:  call       "void System.Console.WriteLine(bool)"
+              IL_0020:  ret
+            }
+            """);
+
+        verifier = CompileAndVerify(new[] { program, interceptor, s_attributesSource }, parseOptions: RegularWithInterceptors, expectedOutput: "True");
+        verifier.VerifyDiagnostics();
+        verifier.VerifyIL("<top-level-statements-entry-point>", """
+            {
+              // Code size       31 (0x1f)
+              .maxstack  2
+              .locals init (MyStruct V_0)
+              IL_0000:  ldloca.s   V_0
+              IL_0002:  initobj    "MyStruct"
+              IL_0008:  ldloc.0
+              IL_0009:  box        "MyStruct"
+              IL_000e:  ldc.i4.1
+              IL_000f:  box        "int"
+              IL_0014:  call       "bool MyInterceptors.Interceptors.Equals(System.ValueType, object)"
+              IL_0019:  call       "void System.Console.WriteLine(bool)"
+              IL_001e:  ret
+            }
+            """);
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/70841")]
+    public void InterceptorStructConstrainedInterfaceMethod()
+    {
+        var program = ("""
+            using System;
+
+            C.M(default(MyStruct));
+
+            class C
+            {
+                public static void M<T>(T t) where T : struct, I
```

Comment:
>struct, [](http://example.com/codeflow?start=55&length=7)

Consider testing without the `struct` constraint as well.

Reply:
Or alternatively, perhaps remove all constraints and test a call to a virtual method on `object`.
```csharp
static void M<T>(T t)
{
    t.ToString();
}
```